lib/reply.js

```diff
@@ -244,6 +249,33 @@ Reply.prototype.headers = function (headers) {
   return this
 }
 
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Trailer#directives
+// https://httpwg.org/specs/rfc7230.html#chunked.trailer.part
+const INVALID_TRAILERS = [
+  'transfer-encoding',
+  'content-length',
+  'host',
+  'cache-control',
+  'max-forwards',
+  'te',
+  'authorization',
+  'set-cookie',
+  'content-encoding',
+  'content-type',
+  'content-range',
+  'trailer'
+]
+
+Reply.prototype.addTrailer = function (key, fn) {
+  const _key = key.toLowerCase()
+  if (INVALID_TRAILERS.includes(_key)) {
```

Comment:
Could `INVALID_TRAILERS` be a `Map` in order to execute a `.has` here?

The array is just internal, so this would simplify the check

Reply:
Is it better if it change to use `Set` instead of `Map` or `Array`?

Reply:
Yeah, the complex time is the same `O(1)` 👍🏽